File: matchingBraces.js
function matchingBraces(braces) {
	const ans = [];
	for (let brace of braces) {
		let match = brace.match(/(\{\}|\(\)|\[\])/);
		while (match) {
			brace = brace.replace(match[0], '');
			match = brace.match(/(\{\}|\(\)|\[\])/);
		}
		if (brace.length > 0) ans.push('NO');
		else ans.push('YES');
	}
	return ans;
}


console.log(matchingBraces(['[{}]', '[{]}', '{([])}']));

File: python/matchingBraces.py
import re

def matchingBraces(braces):
    ans = []
    for brace in braces:
        while '()' in brace or '[]' in brace or '{}' in brace:
            brace = brace.replace('()', '').replace('[]', '').replace('{}', '')
        
        if len(brace) > 0:
            ans.append('NO')
        else:
            ans.append('YES')
    return ans
    
print(matchingBraces(['[{}]','[{]}']))